mm/2s2h/Rando/ActorBehavior/ObjSnowball.cpp

```diff
@@ -0,0 +1,240 @@
+    { { -1343.00f, 776.00f }, RC_TWIN_ISLANDS_SMALL_SNOWBALL_02 },
+    { { 99.00f, -221.00f }, RC_TWIN_ISLANDS_SMALL_SNOWBALL_03 },
+    { { -1349.00f, 114.00f }, RC_TWIN_ISLANDS_SMALL_SNOWBALL_04 },
+    { { -1264.00f, 161.00f }, RC_TWIN_ISLANDS_SMALL_SNOWBALL_05 },
+    { { -1326.00f, 172.00f }, RC_TWIN_ISLANDS_SMALL_SNOWBALL_06 },
+
+};
+
+RandoCheckId IdentifySnowball(Vec3f pos) {
+    RandoCheckId randoCheckId = RC_UNKNOWN;
+
+    auto it = snowballMap.find({ pos.x, pos.z });
+    if (it == snowballMap.end()) {
+        return randoCheckId;
+    }
+
+    return it->second;
+}
+
+void SpawnSnowballDrop(Vec3f pos, RandoCheckId randoCheckId) {
+    CustomItem::Spawn(
+        pos.x, pos.y, pos.z, 0, CustomItem::KILL_ON_TOUCH | CustomItem::TOSS_ON_SPAWN | CustomItem::ABLE_TO_ZORA_RANG,
+        randoCheckId,
+        [](Actor* actor, PlayState* play) {
+            RandoSaveCheck& randoSaveCheck = RANDO_SAVE_CHECKS[CUSTOM_ITEM_PARAM];
+            randoSaveCheck.eligible = true;
+        },
+        [](Actor* actor, PlayState* play) {
+            auto& randoSaveCheck = RANDO_SAVE_CHECKS[CUSTOM_ITEM_PARAM];
+            RandoItemId randoItemId = Rando::ConvertItem(randoSaveCheck.randoItemId);
+            Matrix_Scale(30.0f, 30.0f, 30.0f, MTXMODE_APPLY);
+            Rando::DrawItem(Rando::ConvertItem(randoSaveCheck.randoItemId, (RandoCheckId)CUSTOM_ITEM_PARAM), actor);
+        });
+}
+
+// void ObjSnowball_RandoDraw(Actor* actor, PlayState* play) {
+//     if (!CVarGetInteger("gRando.CSMC", 0)) {
+//         Gfx_DrawDListOpa(play, (Gfx*)gBarrelJunkDL);
+//         return;
+//     }
+//
+//     RandoItemId randoItemId = Rando::ConvertItem(RANDO_SAVE_CHECKS[OBJTARU_RC].randoItemId,
+//     (RandoCheckId)OBJTARU_RC); RandoItemType randoItemType = Rando::StaticData::Items[randoItemId].randoItemType;
+//
+//     switch (randoItemType) {
+//         case RITYPE_BOSS_KEY:
+//             Gfx_DrawDListOpa(play, (Gfx*)gBarrelBossKeyDL);
+//             break;
+//         case RITYPE_HEALTH:
+//             Gfx_DrawDListOpa(play, (Gfx*)gBarrelHeartDL);
+//             break;
+//         case RITYPE_LESSER:
+//             Gfx_DrawDListOpa(play, (Gfx*)gBarrelMinorDL);
+//             break;
+//         case RITYPE_MAJOR:
+//             Gfx_DrawDListOpa(play, (Gfx*)gBarrelMajorDL);
+//             break;
+//         case RITYPE_MASK:
+//             Gfx_DrawDListOpa(play, (Gfx*)gBarrelMaskDL);
+//             break;
+//         case RITYPE_SKULLTULA_TOKEN:
+//             Gfx_DrawDListOpa(play, (Gfx*)gBarrelTokenDL);
+//             break;
+//         case RITYPE_SMALL_KEY:
+//             Gfx_DrawDListOpa(play, (Gfx*)gBarrelSmallKeyDL);
+//             break;
+//         case RITYPE_STRAY_FAIRY:
+//             Gfx_DrawDListOpa(play, (Gfx*)gBarrelFairyDL);
+//             break;
+//         default:
+//             Gfx_DrawDListOpa(play, (Gfx*)gBarrelJunkDL);
+//             break;
+//     }
+// }
```

Comment:
We can just axe this, when we need it we can copy from the other CSMC implementations